src/state/SetLegendPayload.ts

```diff
@@ -33,6 +33,6 @@ export function SetPolarLegendPayload({ legendPayload }: { legendPayload: Readon
     return () => {
       dispatch(removeLegendPayload(legendPayload));
     };
-  }, [dispatch, legendPayload]);
+  }, [dispatch, layout, legendPayload]);
```

Comment:
@PavelVanecek might have omitted this on purpose. Will wait for his review

Reply:
I would have left a comment if it was on purpose. Thanks for the fix

Reply:
You're welcome!